src/nccl_ofi_interface_neuron.c

```diff
@@ -6,6 +6,23 @@
 
 #include "nccl_ofi.h"
 #include "nccl_ofi_api.h"
+#include "nccl_ofi_param.h"
+
+static ncclResult_t init_v4(ncclDebugLogger_t logFunction)
+{
+	/*
+	 * RDMA protocol `connect()` returns a valid send communicator only
+	 * after a connect response message is received from peer. Because the
+	 * v4 net-plugin `connect()` API is expected to synchronously return a
+	 * valid send communicator (a behaviour that was changed since v5+),
+	 * this RDMA protocol behaviour is incompatible with v4 `connect()`
+	 * API.
+	 */
+	if (getenv("OFI_NCCL_PROTOCOL") == NULL) {
+		setenv("OFI_NCCL_PROTOCOL", "SENDRECV", 0);
```

Comment:
Need to error check the setenv here.

And nit: I'm not a huge fan of us setting our own params via `setenv`, because (in `nccl_ofi_param.h`) we store the value of these params on first lookup, so if we happened to call `ofi_nccl_protocol()` before we set this env var, setting the env var will have no effect.

In this case, since we are setting the env var before init, I don't see how this could go wrong. It does feel hacky. It would be better to have getters/setters for the params, but we can do that in a future change.

Reply:
Michael pointed out that at this point, the logger has not been initialized (since it is initialized in `nccl_net_ofi_init`), so `NCCL_OFI_WARN` and family of calls won't work. I suggested my preference for failing with a simple `perror` if `setenv` fails.

A less-favorable (IMO) alternative is to initialize the logger here as well.

Reply:
See my comments above.  We're doing setenv here because the params system isn't initialized yet.  We should fix that, but we don't have to do it now.

Reply:
I also don't think it matters if we fail to `setenv()`. In case it fail, we will use the platform default protocol which is now SENDRECV (works fine) and later will be RDMA and then it will explicitly fail with a proper error message on `connect()` API.

Reply:
Agree. What exactly would you do if setenv failed anyway?

Reply:
My idea was to print a warning. Anyways, let me remote the if check and just do `setenv("OFI_NCCL_PROTOCOL", "SENDRECV", 0);` without error checking. I think that sums the discussion up.

Reply:
No. I think the commit should stay as is. We want that if user sets `OFI_NCCL_PROTOCOL` to RDMA on its own, that the `connect()` API will explicitly fail on that. Rather than we making it work with SENDRECV under the hood which is misleading the user as if the workload works with RDMA protocol when it is not.